frontend/src/scenes/surveys/SurveyAppearancePreview.tsx

```diff
@@ -15,13 +17,15 @@ export function SurveyAppearancePreview({
     const surveyPreviewRef = useRef<HTMLDivElement>(null)
     const feedbackWidgetPreviewRef = useRef<HTMLDivElement>(null)
 
+    const { surveysHTMLAvailable } = useValues(surveysLogic)
+
     useEffect(() => {
         if (surveyPreviewRef.current) {
-            renderSurveysPreview(survey, surveyPreviewRef.current, previewPageIndex)
+            renderSurveysPreview(survey, !surveysHTMLAvailable, surveyPreviewRef.current, previewPageIndex)
```

Comment:
This call doesn't match the function signature in `posthog-js`, likely causing this error:

![image](https://github.com/PostHog/posthog/assets/22996112/f8525a41-55f2-47cd-a839-ef45293c43ee)

I'd recommend using named parameters here, i.e. `renderSurveysPreview({ survey, isSurveysHTMLAvailable, parentElement, previewPageIndex })`

Reply:
oh interesting, I didn't see that error when I ran this, but good shout on the advice.

Reply:
Okay, so how did you imagine using named parameters here?  I tried doing this (in posthog-js)

```
export const renderSurveysPreview = ({
    survey,
    parentElement,
    previewPageIndex,
    forceDisableHtml,
}: {
    survey: Survey
    parentElement: HTMLElement
    previewPageIndex: number
    forceDisableHtml?: boolean
}) => {
    const surveyStyleSheet = style(survey.appearance)
    const styleElement = Object.assign(document.createElement('style'), { innerText: surveyStyleSheet })

    // Remove previously attached <style>
    Array.from(parentElement.children).forEach((child) => {
        if (child instanceof HTMLStyleElement) {
            parentElement.removeChild(child)
        }
    })

    parentElement.appendChild(styleElement)
    const textColor = getContrastingTextColor(
        survey.appearance?.backgroundColor || defaultSurveyAppearance.backgroundColor || 'white'
    )

    Preact.render(
        <SurveyPopup
            key="surveys-render-preview"
            survey={survey}
            forceDisableHtml={forceDisableHtml}
            style={{
                position: 'relative',
                right: 0,
                borderBottom: `1px solid ${survey.appearance?.borderColor}`,
                borderRadius: 10,
                color: textColor,
            }}
            previewPageIndex={previewPageIndex}
        />,
        parentElement
    )
}
```

And then doing changing `SurveyAppearancePreview` to pass in named params
```
        if (surveyPreviewRef.current) {
            renderSurveysPreview({
                survey,
                parentElement: surveyPreviewRef.current,
                previewPageIndex,
                forceDisableHtml: !surveysHTMLAvailable, // If users aren't allowed to use HTML, never render it
            })
        }
```

But that means that if the user is using an outdated version of posthog JS, they won't be able to render the survey at all, right, because they'll just be pass in parameters when posthog-js expects an object. I know that posthog-js needs to be backward compatible, so this feels a little unsafe.  Or is this fine because the rendering methods are only used in `posthog` itself, and therefore will be updated along with the posthog changes?

Reply:
Exactly, `renderSurveysPreview` is only used in `posthog`, and that will always have the latest version of `posthog-js`, so this is safe.

The rendering of the actual live popup is done here: https://github.com/PostHog/posthog-js/blob/main/src/extensions/surveys.tsx#L101

Good shout to consider backwards compatibility :)